src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/service/PniRiskEngineTest.kt

```diff
@@ -136,7 +136,117 @@ class PniRiskEngineTest {
+      ogrs3 = null,
+      ovp = null,
+      ospDc = ospDcValue,
+      ospIic = null,
+      rsr = null,
+      sara = Sara(
+        saraRiskOfViolenceTowardsPartner = null,
+        saraRiskOfViolenceTowardsOthers = null,
+        overallResult = null,
+      ),
+    )
+    assertTrue(riskEngine.isHighRisk(riskScores, "Male"))
+  }
+
+  @ParameterizedTest
+  @CsvSource(value = ["VERY_HIGH", "VERY HIGH", "Very High", "HIGH", "High"])
+  fun `should return isHighRisk for a person with High or Very High Sara risk of violence towards partner`(saraRiskValue: String) {
+    val riskScores = IndividualRiskScores(
+      ogrs3 = null,
+      ovp = null,
+      ospDc = null,
+      ospIic = null,
+      rsr = null,
+      sara = Sara(
+        saraRiskOfViolenceTowardsPartner = saraRiskValue,
+        saraRiskOfViolenceTowardsOthers = null,
+        overallResult = null,
+      ),
+    )
+    assertTrue(riskEngine.isHighRisk(riskScores, "Male"))
+    assertTrue(riskEngine.isHighRisk(riskScores, "Female"))
+  }
+
+  @ParameterizedTest
+  @CsvSource(value = ["VERY_HIGH", "VERY HIGH", "Very High", "HIGH", "High"])
+  fun `should return isHighRisk for a person with High or Very High Sara risk of violence towards others`(saraRiskValue: String) {
+    val riskScores = IndividualRiskScores(
+      ogrs3 = null,
+      ovp = null,
+      ospDc = null,
+      ospIic = null,
+      rsr = null,
+      sara = Sara(
+        saraRiskOfViolenceTowardsPartner = null,
+        saraRiskOfViolenceTowardsOthers = saraRiskValue,
+        overallResult = null,
+      ),
+    )
+    assertTrue(riskEngine.isHighRisk(riskScores, "Male"))
+    assertTrue(riskEngine.isHighRisk(riskScores, "Female"))
+  }
+
+  @ParameterizedTest
+  @CsvSource(value = ["VERY_HIGH", "VERY HIGH", "Very High", "HIGH", "High"])
+  fun `should NOT return isHighRisk for a female with High or Very High ospDc`(ospDcValue: String) {
+    val riskScores = IndividualRiskScores(
+      ogrs3 = null,
+      ovp = null,
+      ospDc = ospDcValue,
+      ospIic = null,
+      rsr = null,
+      sara = Sara(
+        saraRiskOfViolenceTowardsPartner = null,
+        saraRiskOfViolenceTowardsOthers = null,
+        overallResult = null,
+      ),
+    )
+    assertFalse(riskEngine.isHighRisk(riskScores, "Female"))
+  }
+
+  @ParameterizedTest
+  @CsvSource(value = ["VERY_HIGH", "VERY HIGH", "Very High", "HIGH", "High"])
+  fun `should NOT return isHighRisk for a female with High or Very High ospIic`(ospIicValue: String) {
+    val riskScores = IndividualRiskScores(
+      ogrs3 = null,
+      ovp = null,
+      ospDc = null,
+      ospIic = ospIicValue,
+      rsr = null,
+      sara = Sara(
+        saraRiskOfViolenceTowardsPartner = null,
+        saraRiskOfViolenceTowardsOthers = null,
+        overallResult = null,
+      ),
+    )
+    assertFalse(riskEngine.isHighRisk(riskScores, "Female"))
+  }
+
+  @ParameterizedTest
+  @CsvSource(value = ["Female,false", "Male,true"], delimiter = ',')
```

Comment:
NB: Interesting to note that previously the combination of "Female, true" gave a positive result previously here which is incorrect IMHO.

Reply:
Not sure I fully understand this, but its best checking the logic with Alex.